heap/max_heap.py

```diff
@@ -1,21 +1,45 @@
 class Heap:
-  def __init__(self):
```

Comment:
I'll try to review this when I have the chance; tbh, I'm weak on heaps. I need to do some review on this concept; will get back to this. Great thing is, once you got max heap, you can easily write out a min heap, or a generalized heap!